saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java

```diff
@@ -304,6 +305,16 @@ public static InputStream getSourceAsStream(Source source) throws ConfigurationE
         Result streamResult = new StreamResult(baos);
         // Write the DOM document to the stream
         Transformer transformer = TransformerUtil.getTransformer();
+
+        if (DOMSource.class.isInstance(source)) {
+            Node node = ((DOMSource) source).getNode();
+            if (Document.class.isInstance(node)) {
+                String xmlEncoding = ((Document) node).getXmlEncoding();
+                if (xmlEncoding != null)
+                    transformer.setOutputProperty(OutputKeys.ENCODING, xmlEncoding);
```

Comment:
I'd rather suggest keeping TransformerUtil as is and including the xml declaration if necessary due to encoding difference. This is few bytes difference yet that can be helpful in redirect binding.
```suggestion
                    transformer.setOutputProperty(OutputKeys.ENCODING, xmlEncoding);
                    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "no");
```

Reply:
done